torch_xla/csrc/ops/binary_cross_entropy.cpp

```diff
@@ -0,0 +1,74 @@
+#include "torch_xla/csrc/ops/binary_cross_entropy.h"
+
+#include "tensorflow/compiler/xla/xla_client/util.h"
+#include "tensorflow/core/lib/gtl/array_slice.h"
+#include "torch_xla/csrc/lowering_context.h"
+#include "torch_xla/csrc/ops/infer_output_shape.h"
+
+namespace torch_xla {
+namespace ir {
+namespace ops {
+namespace {
+
+xla::Shape NodeOutputShape(const Value& logits, const Value& labels,
+                           const absl::optional<Value>& weight,
+                           ReductionMode reduction) {
+  auto lower_for_shape_fn =
+      [&](tensorflow::gtl::ArraySlice<const xla::XlaOp> operands)
+      -> xla::XlaOp {
+    absl::optional<xla::XlaOp> weight;
+    if (operands.size() > 2) {
+      weight = operands[2];
+    }
+    return BuildBinaryCrossEntropy(operands[0], operands[1], weight, reduction);
+  };
+  std::vector<xla::Shape> shapes;
+  for (auto& input :
+       xla::util::GetValuesVector<Value>({logits, labels}, {&weight})) {
+    shapes.push_back(input.shape());
+  }
+  return InferOutputShape(shapes, lower_for_shape_fn);
+}
+
+}  // namespace
+
+BinaryCrossEntropy::BinaryCrossEntropy(const Value& logits, const Value& labels,
+                                       const absl::optional<Value>& weight,
+                                       ReductionMode reduction)
+    : Node(ir::OpKind(at::aten::binary_cross_entropy),
+           xla::util::GetValuesVector<Value>({logits, labels}, {&weight}),
+           [&]() { return NodeOutputShape(logits, labels, weight, reduction); },
+           /*num_outputs=*/1,
+           xla::util::MHash(xla::util::GetEnumValue<ReductionMode>(reduction))),
```

Comment:
No need for template specification. Posting a PR to remove it from other places as well.